neurom/core/types.py

```diff
@@ -44,15 +44,22 @@ class NeuriteIter(OrderedEnum):
     NRN = 2
 
 
+# Adding 100 to SWC actual values
+# to force people to use the Enum
+# and not rely on the fact that
+# the Enum value matches the SWC one
+_INTERFACE_DECOUPLING = 100
```

Comment:
Seriously? What about cells with types that do not correspond to the swc ones because the cells are not neurons? I don't understand why everybody has to use the Neuron types and create maps. What is wrong with things being intuitive in `NeuroM`...

Reply:
I don't get what is this offset? Also if this comment is correct " # the Enum value matches the SWC one " it doesn't make much sense.

Reply:
>Seriously? What about cells with types that do not correspond to the neurom ones because the cells are not neurons?

The point of this is to encourage the use of the named types, as opposed to just using integers to identify different neurite types.  By using the names values, it makes the code more readable, searchable and thus maintainable.

> I don't understand why everybody has to use the Neuron types and create maps.

I'm not sure I follow.  What maps are people creating?

Reply:
2 things:
- Enum values are not exposed to the user and should NOT be used. Never. So, you should not care about them. It could be 47, 52 and -27, NeuroM would work the same. NeuroM is a neuron loader, it should not be attached to a given file format. What if we start supporting other file formats that have different numbering conventions ?
- The problem @eleftherioszisis mentionned does not stem in the NeuriteType enum values but in the fact that non neuron cells have section type of (python) type NeuriteType. This is a more valid debate to me.

Reply:
Yeah, to second @mgeplf, if you have use cases were you have to use integer value because Enums don't work out, brings them here so we can discuss them.

Reply:
> I don't get what is this offset? Why should we not conform to the swc types?

We do conform to SWC types. When you load a section = 2 in NeuroM you end up with NeuriteType.axon.
It is just that the internal values (which you should never use) do not map to SWC values. It's done on purpose to prevent people to use this internal values.

Reply:
When I use the `neurom`, `tmd`, `tns`, `morphio` tools I have to create a mapping from an integer file section type to all of these tools. So far, the last two agree, but `neurom`, which doesn't conform to the swc numbering, has to have a separate treatment. So instead of having a unified logic, I have to take cases for the arbitrary types. I don't understand why do we have to punish the users that for any reason would like to use the underlying types.

Furthermore, when i process cells, like astrocytes, I usually need to create intermediate types, from which I need to extract stats with `NeuroM`. This is why for example undefined or custom neurite types are useful. It is cleaner in my opinion to be able to say `endfoot -> 2` and expect `NeuroM` to map a neurite with type 2, instead of `endfoot -> 2 -> neuron axon -> neurom interface -> 102`

Reply:
In my opinion, having different loaders might not generalize optimally. I have cells for various different types (microglia, oligodendrocytes etc) that obey the swc standards. On the other hand, I also have neurons from different labs with all sorts of "weird" neurite types. I'm not sure what's the best way to deal with all these inconsistencies, but separating loaders might not be the optimal strategy.

Reply:
It has to be said however that different loaders make sense for cells that carry additional datasets that are unique to their type. For example, astrocytes have perimeters that do not exist in neuron moprhs.